src/items/generics.md

```diff
@@ -227,6 +227,22 @@ fn generic<const B: bool>() {
 }
 ```
 
+r[items.generics.const.inferred]
+
+r[items.generics.const.inferred.syntax]
+```grammar,const
+InferredConst ->
+      `_`
+    | `(` InferredConst `)`
+```
+
+r[items.generics.const.inferred.intro]
+The inferred const asks the compiler to infer the const argument if possible
+based on the surrounding information available.
+
+r[items.generics.const.inferred.constraint]
+It cannot be used in item signatures.
```

Comment:
I'm not quite following this. If we are defining this as a new form allowed in generic *args*, why would there be a rule that it is not allowed in item signatures?

Reply:
We don't allow type inference in signatures in general, not just as a restriction on const arguments. For example this is an error on stable:
```rust
struct Foo<T>(T);
fn bar(param: Foo<_>) {}
```

Reply:
Isn't this already stated in type.inferred.constraint (inferred.md)?

Reply:
Yeah we state that for `_`  type arguments but I felt like like we should note that the restriction applies for `_` const arguments too when discussing them as it might be confusing otherwise; people might get the idea that you *can* use `_` const arguments in signatures.